src/confluent_kafka/src/Consumer.c

```diff
@@ -1575,6 +1575,8 @@ static void Consumer_rebalance_cb (rd_kafka_t *rk, rd_kafka_resp_err_t err,
 		if (result)
 			Py_DECREF(result);
 		else {
+			
+			CallState_fetch_exception(cs);
```

Comment:
There are other places as well where we need to handle this. For example `Consumer_offset_commit_cb`. Please check other places where this is needed.